src/hdr_histogram.c

```diff
@@ -343,12 +343,14 @@ static int32_t buckets_needed_to_cover_value(int64_t value, int32_t sub_bucket_c
 int hdr_calculate_bucket_config(
```

Comment:
Can we have API `hdr_calculate_bucket_config2` which calls `hdr_calculate_bucket_config` .
I see the following changes: 

- significant_figures  is passed as  1
- sub_bucket_count_magnitude is calculated based on the yb_bucket_factor --> can this be encapsulated out

Reply:
Made change to address this. Simplified case statement for yb_bucket_factor calculating into sub_bucket_count_magnitude.